Workaround for API change in >=glib-2.16.1, bug #213382

This is not the final solution, but the patches posted on the mailing-list
cause more breakage than they help to solve the problem.

Index: pinentry/gtk+-2/gtksecentry.c
===================================================================
--- pinentry/gtk+-2/gtksecentry.c	(revision 181)
+++ pinentry/gtk+-2/gtksecentry.c	(working copy)
@@ -270,7 +270,11 @@
 
 
 gpointer
+#if GLIB_CHECK_VERSION(2, 16, 1)
+g_malloc(gsize size)
+#else
 g_malloc(gulong size)
+#endif
 {
     gpointer p;
 
@@ -288,7 +292,11 @@
 }
 
 gpointer
+#if GLIB_CHECK_VERSION(2, 16, 1)
+g_malloc0(gsize size)
+#else
 g_malloc0(gulong size)
+#endif
 {
     gpointer p;
 
@@ -308,7 +316,11 @@
 }
 
 gpointer
+#if GLIB_CHECK_VERSION(2, 16, 1)
+g_realloc(gpointer mem, gsize size)
+#else
 g_realloc(gpointer mem, gulong size)
+#endif
 {
     gpointer p;
 
